### PR TITLE
[Chore](hash-join) remove useless conditions and add some case

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes

remove useless conditions and add some case

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

File: be/src/vec/exec/join/vhash_join_node.cpp [hunk 1]
```diff
@@ -791,11 +791,10 @@ Status HashJoinNode::_materialize_build_side(RuntimeState* state) {
 
             RETURN_IF_ERROR(sink(state, &block, eos));
         }
-        RETURN_IF_ERROR(child(1)->close(state));
     } else {
-        RETURN_IF_ERROR(child(1)->close(state));
         RETURN_IF_ERROR(sink(state, nullptr, true));
     }
+    RETURN_IF_ERROR(child(1)->close(state));
```
Comment (mrhhsg) [inline be/src/vec/exec/join/vhash_join_node.cpp:797]: this change will make child(1) close after `sink`

**Review by Gabriel39 — DISMISSED** [review]
LGTM

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Pull request merged**